shared/util/quit-helper.desktop.js

```diff
@@ -27,17 +30,23 @@ function isMainThread () {
 }
 
 function _executeActions (actions: Array<Action>) {
+  // Don't allow us to re-enter this logic if we're already shutting down
+  if (isQuitting) {
```

Comment:
maybe a try/catch around `quit()` below just in case? where a catch will reset isQuitting to false